src/db.rs

```diff
@@ -146,47 +207,78 @@ pub fn get_secret_policy(sec: &str) -> Option<policy::Policy> {
 
-    let mqstr = "INSERT INTO secpol (secret, polids ) VALUES(?, ?)".to_string();
+    let mysqlstr = "INSERT INTO keysets (keysetid, kskeys, polid)
+                    VALUES(?, ?, ?)"
+        .to_string();
 
     let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
+    let kskeys_str = serde_json::to_string(kskeys)?;
 
-    trnsx.exec_drop(mqstr, (&sec, &polid))?;
-    trnsx.commit()?;
-    Ok(polid)
+    trnsx.exec_drop(mysqlstr, (ksetid, &kskeys_str, polid))?;
+
+    Ok(trnsx.commit()?)
 }
 
-pub fn delete_polid(sec: &str) -> Result<String> {
-    let mut dbconn = get_dbconn().unwrap();
+pub fn delete_keyset(ksetid: &str) -> Result<()> {
+    let mut dbconn = get_dbconn()?;
 
-    let mqstr = "DELETE from secpol WHERE secret = ?";
+    let mysqlstr = "DELETE from keysets WHERE keysetid = ?".to_string();
 
     let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
 
-    trnsx.exec_drop(mqstr, (&sec,))?;
-    trnsx.commit()?;
-    Ok(sec.to_string())
+    trnsx.exec_drop(mysqlstr, (ksetid,))?;
+
+    Ok(trnsx.commit()?)
 }
 
-pub fn get_keyset_policy(_secset: &str) -> Option<policy::Policy> {
-    None
+pub fn get_keyset_policy(secset: &str) -> Result<policy::Policy> {
+    let mut dbconn = get_dbconn()?;
+    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+
+    let mqstr = "SELECT polid FROM keysets WHERE keysetid = ?";
+    let val: Option<u32> = trnsx.exec_first(mqstr, (secset,))?;
+
+    let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = ?";
+
+    let polval: Vec<(String, String, u32, u32, String)> = trnsx.exec(mqstr, (val,))?;
+
+    Ok(policy::Policy {
+        allowed_digests: serde_json::from_str(&polval[0].0).unwrap(),
+        allowed_policies: serde_json::from_str(&polval[0].1).unwrap(),
+        min_fw_api_major: polval[0].2,
+        min_fw_api_minor: polval[0].3,
+        allowed_build_ids: serde_json::from_str(&polval[0].4).unwrap(),
+    })
 }
 
-// A keyset is just a group of keys.
-pub fn get_keyset_ids(_id: &str) -> Result<Vec<String>> {
-    Ok(vec!["keyid-1".to_string()])
+// A keyset is just a group of keys. This function returns a vector of key ids for secrets in table secrets
+pub fn get_keyset_ids(id: &str) -> Result<Vec<String>> {
+    let mut dbconn = get_dbconn()?;
+    let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+
+    let mqstr = "SELECT kskeys FROM keysets WHERE keysetid = ?";
+    let retrow: mysql::Row = trnsx.exec_first(mqstr, (id,))?.unwrap();
+
+    let retkeyset = retrow["kskeys"].clone();
+    let kskeysval = match mysql::from_value_opt::<String>(retkeyset) {
+        Ok(string) => string,
+        Err(mysql::FromValueError(_unknown_val)) => {
+            "unknown value returned from from_value_opt".to_string()
+        }
+    };
```

Comment:
there must be a nicer way to get a single string from the result.

Reply:
Fixed.